CMakeLists.txt

```diff
@@ -158,117 +263,97 @@ if (NOT MSVC)
                     set(llhttp_target llhttp_static)
                 endif()
             endif()
-        endif()
-        set(http_lib "-lllhttp")
-
-        if (NOT Jsoncpp_FOUND)
-            message(SEND_ERROR "Jsoncpp is required for DHT proxy support")
-        endif()
-        if (OPENDHT_PROXY_OPENSSL)
-            # https://cmake.org/cmake/help/latest/module/FindOpenSSL.html
-            pkg_search_module(OPENSSL REQUIRED IMPORTED_TARGET openssl)
-            if (OPENSSL_FOUND)
-                message(STATUS "Found OpenSSL ${OPENSSL_VERSION} ${OPENSSL_INCLUDE_DIRS}")
-                set(openssl_lib ", openssl")
-            else ()
-                message(SEND_ERROR "OpenSSL is required for DHT proxy as specified")
+            set(http_lib "-lllhttp")
+            
+            if (NOT Jsoncpp_FOUND)
+                message(SEND_ERROR "Jsoncpp is required for DHT proxy support")
             endif()
-        endif()
+            if (OPENDHT_PROXY_OPENSSL)
+                # https://cmake.org/cmake/help/latest/module/FindOpenSSL.html
+                pkg_search_module(OPENSSL REQUIRED IMPORTED_TARGET openssl)
+                if (OPENSSL_FOUND)
+                    message(STATUS "Found OpenSSL ${OPENSSL_VERSION} ${OPENSSL_INCLUDE_DIRS}")
+                    set(openssl_lib ", openssl")
+                else ()
+                    message(SEND_ERROR "OpenSSL is required for DHT proxy as specified")
+                endif()
+            endif()
+        else ()
+            set(OPENDHT_PROXY_OPENSSL OFF)
+        endif ()
+    endif ()
+else () # MSVC
+    set(CMAKE_CXX_FLAGS "/utf-8 /EHsc") # Unicode workaround
+    if (BUILD_SHARED_LIBS)
+        message(STATUS "Building shared library")
+        set(VCPKG_TARGET_TRIPLET "x64-windows")
```

Comment:
at this stage, vcpkg dependencies are already built/installed (vcpkg build runs at the beginning of the configure logic).
Maybe we should let the caller set these

Reply:
Fair enough, let's remove this :smile:

Reply:
69f23e49c47feafa86b5fceadb1bcb4339f9f610